models/demos/t3000/mixtral8x7b/tests/test_mixtral_model_prefill.py

```diff
@@ -0,0 +1,138 @@
+    os.environ["MIXTRAL_CACHE_PATH"] = "/mnt/MLPerf/tt_dnn-models/Mistral/Mixtral-8x7B-v0.1/"
+    os.environ["TT_METAL_ASYNC_DEVICE_QUEUE"] = "1"
+    os.environ["WH_ARCH_YAML"] = "wormhole_b0_80_arch_eth_dispatch.yaml"
+
+import ttnn
+from ttnn import ReplicateTensorToMesh, ConcatMeshToTensor
+
+from models.demos.t3000.mixtral8x7b.tt.mixtral_common import (
+    prepare_inputs_ttnn_prefill,
+    prepare_rotation_mat_ttnn,
+    get_rot_transformation_mat,
+)
+from models.demos.t3000.mixtral8x7b.tt.mixtral_model import TtTransformer
+from models.demos.t3000.mixtral8x7b.reference.model import Transformer
+from models.demos.t3000.mixtral8x7b.reference.tokenizer import Tokenizer
+from models.demos.t3000.mixtral8x7b.tt.model_config import TtModelArgs
+from models.utility_functions import comp_pcc, comp_allclose
+
+
+class Emb(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.emb = torch.nn.Embedding(32000, 4096)
+
+    def forward(self, x):
+        return self.emb(x)
+
+
+@pytest.mark.parametrize(
+    "seq_len",
+    (128, 1024, 2048),
+)
+@pytest.mark.parametrize(
+    "n_layers",
+    (32,),
+)
+def test_mixtral_model_inference(t3k_device_mesh, use_program_cache, reset_seeds, n_layers, seq_len):
+    pcc = 0.96
+    dtype = ttnn.bfloat8_b
+
+    model_args = TtModelArgs(t3k_device_mesh.get_device(0))
+    model_args.n_layers = n_layers
+    batch = 1
+    state_dict = model_args.load_state_dict()
+
+    tokenizer = Tokenizer(model_args.tokenizer_path)
+    prompt = "Once upon a time, in a charming countryside, there lived three little pigs named Porky, Petunia, and Percy. They were siblings and loved to play together all day long. But one day, their mother knew it was time for them to build their own homes and be independent. Remember, my little ones, their mother said, the world can be tricky, so build your houses strong and sturdy to keep you safe from harm. With hugs and kisses, the three little pigs bid farewell to their mother and set off on their journey to find the perfect spot to build their homes. Porky, being the laziest of the bunch, quickly found a pile of straw nearby and decided it was the perfect place to build his house. With little effort, he constructed a cozy straw house and declared, I'm done! Now I can relax and play all day. Petunia was a bit more hardworking. She found a bunch of sticks and twigs and began building her house. It took a bit longer, but she managed to create a charming little house. Percy, the wisest of the three, knew that hard work pays off. He searched for the sturdiest materials he could find and finally decided on bricks. He carefully stacked and cemented the bricks together, creating a strong and reliable house. One evening, as the sun was setting, a big bad wolf happened upon the three little pigs. He was hungry and had his eyes set on the tasty pigs. The wolf first came across Porky's straw house."
+    prompt = prompt * 7
```

Comment:
This line is to make sure we have a prompt with more than 2k tokens, right?

Would probably be good to update this to a real 2k+ prompt. Ideally just reading from one of the prompts we use in demo prefill.